scripts/validate_docstrings.py

```diff
@@ -331,6 +336,41 @@ def parameter_mismatches(self):
 
         return errs
 
+    @property
+    def pep8_violations(self):
+        with self._file_representation() as file:
+            application = Flake8()
+            application.initialize(["--doctests", "--quiet"])
+            application.run_checks([file.name])
+            application.report()
+            stats = application.guide.stats
+            return [
+                "{} {} {}".format(s.count, s.error_code, s.message)
+                for s in stats.statistics_for('')
+            ]
+
+    @contextmanager
+    def _file_representation(self):
```

Comment:
This name is not very descriptive. I think this is the examples as temporary file, so something like `_examples_as_temp_file` seems much clearer.

Not sure if it's worth having two separate methods, I'd generate the file in the method that validates pep8. But if we have two methods, I'd have this first, as this is being used by the other (`pep8_violations`).